contract/src/main/kotlin/com/r3/corda/sdk/token/contracts/states/AbstractToken.kt

```diff
@@ -1,20 +1,36 @@
 package com.r3.corda.sdk.token.contracts.states
 
+import com.r3.corda.sdk.token.contracts.types.IssuedTokenType
+import com.r3.corda.sdk.token.contracts.types.TokenType
 import net.corda.core.contracts.ContractState
 import net.corda.core.crypto.toStringShort
 import net.corda.core.identity.AbstractParty
 import net.corda.core.identity.AnonymousParty
 import net.corda.core.identity.Party
 
 /** Contains common [NonFungibleToken] functionality. */
-abstract class AbstractToken : ContractState {
+abstract class AbstractToken<T : TokenType> : ContractState {
```

Comment:
This also could be converted into an interface. Would perhaps need to move `holderString` out somewhere else (extension, etc.).

Reply:
Yup fair point - I can do that.